libraries/libfc/test/test_bls.cpp

```diff
@@ -280,5 +246,33 @@ BOOST_AUTO_TEST_CASE(bls_bad_sig_verif) try {
 
 } FC_LOG_AND_RETHROW();
 
+//test private key base58 encoding
+BOOST_AUTO_TEST_CASE(bls_private_key_string_encoding) try {
+
+  bls_private_key sk = bls_private_key(seed_1);
+
+  bls_public_key pk = sk.get_public_key();
+  
+  std::string priv_base58_str = sk.to_string();
+
+  //cout << "priv_base58_str : " << priv_base58_str << "\n";
+
+  bls_private_key sk2 = bls_private_key(priv_base58_str);
+
+  //cout << "sk2 : " << sk2.to_string() << "\n";
+
+  bls_signature signature = sk2.sign(message_1);
+
+  //cout << "pk : " << pk.to_string() << "\n";
+  //cout << "signature : " << signature.to_string() << "\n";
+
+  // Verify the signature
+  bool ok = verify(pk, message_1, signature);
+
+  BOOST_CHECK_EQUAL(ok, true);
+
+} FC_LOG_AND_RETHROW();

```

Comment:
Can you add tests for to/from string for public key and signature as well.

Reply:
There is a test called "bls_serialization_test" which should provide adequate coverage for these, but let me know if it can be improved.